client/src/pages/metadata-query/__tests__/freq-query.test.tsx

```diff
@@ -0,0 +1,79 @@
+import { describe, expect, test, vi } from 'vitest';
+import { render, screen, fireEvent} from '@testing-library/react';
+
+import React from 'react';
+
+import FreqQuery from '@/pages/metadata-query/freq-query';
+
+describe('Test FreqQuery', () => {
+  const mockName = 'frequency name';
+  const mockDescription = 'frequency description';
+
+  test('Basic Rendering', () => {
+    const validatorMock = vi.fn(() => 'this is valid');
+    const handleQueryValidMock = vi.fn();
+    const handleQueryInvalidMock = vi.fn();
+    render(
+      <FreqQuery
+        queryName={mockName}
+        description={mockDescription}
+        validator={validatorMock}
+        handleQueryValid={handleQueryValidMock}
+        handleQueryInvalid={handleQueryInvalidMock}
+      />
+    );
+    expect(screen.getByText('frequency name')).exist;
+  });
+
+  test('Changing the upper frequency bound calls the validator function and handleQueryValid function', () => {
+    const validatorMock = vi.fn(() => 'this is valid');
+    const handleQueryValidMock = vi.fn();
+    const handleQueryInvalidMock = vi.fn();
+    render(
+      <FreqQuery
+        queryName={mockName}
+        description={mockDescription}
+        validator={validatorMock}
+        handleQueryValid={handleQueryValidMock}
+        handleQueryInvalid={handleQueryInvalidMock}
+      />
+    );
+    expect(validatorMock).toHaveBeenCalledWith({
+      from: 30000000,
+      to: 300000000,
+    });
+    const upperBound = screen.getByTestId('freq-upper');
+    fireEvent.change(upperBound, { target: { value: '100000000' } });
+    expect(validatorMock).toHaveBeenCalledWith({
+      from: 30000000,
+      to: '100000000',
+    });
+    expect(handleQueryValidMock).toHaveBeenCalledTimes(1);
+  });
+
+  test('Changing the upper frequency bound to lower than the lower bound calls the validator function and handleQueryInvalid function', () => {
+    const validatorMock = vi.fn(() => false);
+    const handleQueryValidMock = vi.fn();
+    const handleQueryInvalidMock = vi.fn();
+    render(
+      <FreqQuery
+        queryName={mockName}
+        description={mockDescription}
+        validator={validatorMock}
+        handleQueryValid={handleQueryValidMock}
+        handleQueryInvalid={handleQueryInvalidMock}
+      />
+    );
+    expect(validatorMock).toHaveBeenCalledWith({
+      from: 30000000,
+      to: 300000000,
+    });
+    const upperBound = screen.getByTestId('freq-upper');
+    fireEvent.change(upperBound, { target: { value: '1' } });
+    expect(validatorMock).toHaveBeenCalledWith({
+      from: 30000000,
+      to: '1',
+    });
+    expect(handleQueryInvalidMock).toHaveBeenCalledTimes(1);
+  })
```

Comment:
Pls no

Reply:
Couold you provide any reason whhy or a viable alternative ?

Reply:
How does calling a method prove the functionality we want is done. Test the return you see in the page rather than that. I.e is the button still disabled vs is it not

Reply:
The problem it is that for these componenets there is no screen information for this method the only thing that this does it is to return the value to the query page that them use those values to compose the query because of that testing tdirect the method that have been called we could test if the method was called with the right arguments like I chamged in the date query. Idealy this component could be done as a hook that way we could subscribe top the state that is stored and by that we could inspect the state that is done inside this object instead of proping up like that.

Reply:
But this would not be a test of this component this component knows nothing about where it is used it has no knowledge of the query page at all it doesn't even know that one exists.

I agree that a test on the query page that does validate that in case you input a valid date the query button is enable but them if you put an invalid date it is disabled would be useful but that was not the intent of this test file the intent of this test file was to test this component in isolation outside of the context of the query page.

We might get to the conclusion that there is no need to test the component in isolation. So this test will be deemed as obsolete and could be removed.